[hjungwoo01] iP

hjungwoo01 opened the pull request:
# TaskManager Pro

> "The only way to get a thing done is to start." – Langston Hughes ([source](https://www.goodreads.com/quotes/18166-the-only-way-to-get-a-thing-done-is-to#:~:text=Quote%20by%20Langston%20Hughes%3A%20%E2%80%9C.,is%20to%20start...%E2%80%9D))

**TaskManager Pro** is your ultimate companion for task management. It's designed to be:

- **Text-based:** Direct interaction with your tasks through simple text commands.
- **User-friendly:** Intuitive to use right from the start.
- **Efficient:** Quick to add, remove, and manage tasks, making your to-do list a breeze to handle.

**Getting Started:**

1. **Download** the latest version from the [releases page](https://github.com/hjungwoo01/ip/releases/tag/v0.1).
2. **Run** `ip.jar` by double-clicking it or through the command line.
3. **Enjoy** organizing your tasks more efficiently than ever before! 🌟

And yes, it's **completely free**!

## Features

- [x] Task management
- [ ] Deadline tracking 
- [ ] Reminders (coming soon!)

TaskManager Pro isn't just for managing tasks; it's also a great way for Java enthusiasts to **practice Java**. Here's the entry point:

```java
public class Duke {
    public static void main(String[] args) {
        new Duke().run();
    }
}
```

Whether you're a command-line veteran or a Java developer, TaskManager Pro is tailored for you. 

Let's not let ~~procrastination~~ keep us from being productive. Embrace **TaskManager Pro** and make task management a delight!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

hjungwoo01 committed:
Rename chatbot, add user greeting, and implement program exit feature

hjungwoo01 committed:
Add echo feature to the chatbot program

hjungwoo01 committed:
Add 'list' feature to display previously entered user responses

hjungwoo01 committed:
Implemented Task class and added mark/unmark task functionality

hjungwoo01 committed:
Implemented Todo, Event, Deadline functionalities

hjungwoo01 committed:
Added automated text UI testing

hjungwoo01 committed:
Added custom DukeException handler class for handling unusual user inputs

hjungwoo01 committed:
Added task deletion feature

hjungwoo01 committed:
Implemented task history saving and loading from previous sessions

hjungwoo01 committed:
Merge Level 7 Implementation into master

hjungwoo01 committed:
Implement LocalDateTime in Deadline and Event classes

Enhance Deadline and Event classes for date-time handling

Both Deadline and Event classes were previously using String types
for date and time management. This was inconsistent and inefficient
for handling date-time operations.

This commit:
- Modifies the Deadline class to parse and store dates as LocalDateTime
  objects instead of strings, aligning with Java's modern date-time API.
- Updates the Event class similarly, handling both start and end
  times as LocalDateTime instances.
- Refactors the loadTasks method in FileHandler to parse date-time strings
  into LocalDateTime objects for creating Deadline and Event instances.

These changes ensure a consistent and reliable approach to date-time
handling across the application.

hjungwoo01 committed:
Merge branch-Level-8 into master

hjungwoo01 committed:
Refactor codebase for enhanced OOP structure

Implement more OOP principles in project architecture

The previous implementation had intermingled functionalities in a few classes, which made the code less modular and harder to maintain.

This commit:
- Introduces the Ui class to handle all user interactions, encapsulating all UI-related logic.
- Creates the Storage class to manage file operations, specifically for loading and saving tasks. This separation of concerns improves the readability and maintainability of file handling logic.
- Develops the Parser class to interpret user commands, making the command parsing logic more modular and testable.
- Establishes the TaskList class containing and managing the task list, providing a more focused approach to task management.
- Revises the Duke main class to utilize these new classes, significantly decluttering its structure and improving overall code quality.

Furthermore, as part of the stretch goal:
- Command classes such as AddCommand, DeleteCommand, and ExitCommand are extracted. These classes inherit from an abstract Command class, allowing for a more scalable and extensible command handling system in the application.

These modifications align the project more closely with Object-Oriented Programming principles, enhancing code readability, maintainability, and scalability.

hjungwoo01 committed:
Refactor project structure with package organization

Organize classes into structured Java packages

The previous architecture had all classes placed in the default package, which, while functional, did not fully utilize the organizational benefits of Java packages. As the codebase grows, a more structured package organization becomes crucial for maintainability and clarity.

This commit:
- Organizes all classes into a dedicated `duke` package, enhancing project structure.
- Further segregates classes into sub-packages:`duke.task` and `duke.command`. This categorization aligns with each class's functionality.
- Places task-related classes such as `Task`, `Todo`, `Deadline`, and `Event` into `duke.task`.
- Groups command processing classes like `Command`, `AddCommand`, `DeleteCommand`, etc., into `duke.command`.

The `duke.Duke` main class is also updated to reflect these changes.

By arranging classes into distinct packages, the project now benefits from improved organization, easier navigation, and better encapsulation of related functionalities.

These changes are a step towards aligning the project more closely with established Java development practices, enhancing the overall design and scalability of the application.

hjungwoo01 committed:
Merge branch-A-Packages into master

hjungwoo01 committed:
Merge Gradle support into master

hjungwoo01 committed:
Add specific JUnit tests for AddCommand and Parser functionality

In this commit, we introduce targeted unit tests for two key components of our application, the AddCommand and Parser, enhancing the reliability and maintainability of our codebase.

Key Implementations:
1. AddCommandTest: This test class focuses on the functionality of the AddCommand. It includes tests for handling empty descriptions in 'todo' commands and validating the behavior of 'deadline' commands with both valid and invalid date-time formats. The tests ensure that AddCommand correctly processes valid inputs and throws DukeExceptions for invalid or incomplete inputs.

   - addTodo_emptyDescription_throwsDukeException: Verifies that an empty description in a 'todo' command correctly triggers a DukeException.
   - addDeadline_validDateTime_addsDeadline: Confirms that a valid 'deadline' command correctly adds a Deadline task to the TaskList.
   - addDeadline_invalidDateTime_throwsDukeException: Ensures that an invalid date-time format in a 'deadline' command results in a DukeException.

2. ParserTest: This test class assesses the Parser's ability to accurately parse 'todo' and 'deadline' commands into respective AddCommand objects.

   - parse_validTodoCommand_returnsAddCommand: Checks that the Parser correctly interprets a valid 'todo' command.
   - parse_validDeadlineCommand_returnsAddCommand: Verifies that the Parser accurately parses a valid 'deadline' command.

By implementing these tests, we significantly bolster the robustness of our command parsing and task addition processes. This initiative is a step forward in ensuring that our application behaves as expected under various input scenarios and correctly handles edge cases.

Note: The tests employ stubbing of dependencies where necessary to focus solely on the class under test, providing clear, isolated validation of their respective functionalities.

hjungwoo01 committed:
Merge branch-A-JUnit to master

hjungwoo01 committed:
Package app as executable JAR

hjungwoo01 committed:
Merge branch-A-Jar

hjungwoo01 committed:
Add comprehensive documentation and comments to codebase

In this commit, comprehensive documentation and comments have been introduced across the codebase, with a focus on improving code readability, maintainability, and understandability. The key enhancements include:

1. Javadoc comments have been added for classes, methods, and fields, providing detailed descriptions of their purpose, parameters, and return values. This documentation serves as a valuable resource for developers working on the project and for those seeking to understand the code.
2. In-code comments have been included where necessary to explain complex algorithms, data structures, and logic. These comments aim to make the code more transparent and assist developers in understanding the intricacies of the implementation.
3. Naming conventions have been improved by refining variable and method names to adhere to best practices and follow a consistent naming convention throughout the codebase. This consistency enhances code clarity and reduces the likelihood of naming-related errors.
4. Enhanced error handling documentation provides detailed explanations for error messages and exceptions, making it easier for developers to diagnose issues and troubleshoot errors effectively.

These documentation and commenting improvements aim to create a more accessible and developer-friendly codebase, facilitating collaboration, code maintenance, and the onboarding of new team members. Ultimately, these enhancements contribute to the overall quality and sustainability of the software project.

hjungwoo01 committed:
Merge branch-A-JavaDoc to master

hjungwoo01 committed:
Refactor codebase for compliance with coding standard

In this commit, the codebase has been refactored to align with the prescribed coding standard. Going forward, any new code additions will adhere strictly to the same coding standard, ensuring consistency and maintainability throughout the project.

hjungwoo01 committed:
Merge branch-A-CodingStandard to master

hjungwoo01 committed:
Add FindCommand to search for tasks by keyword

This commit introduces a new functionality, `FindCommand`, to the Duke application. The `FindCommand` allows users to search for tasks containing a specific keyword in their descriptions. The following changes have been made:

- Added `FindCommand.java` to represent the command for task searching.
- Implemented the execution logic for `FindCommand`, which filters tasks based on the provided keyword and displays matching tasks to the user.
- Ensured proper error handling for cases where the keyword is missing.
- Followed coding standards for commit message format, subject length, and imperative mood.

With this new feature, users can easily find and retrieve tasks that match their search criteria, enhancing the usability of the Duke application.

hjungwoo01 committed:
Merge branch-Level-9 to master

hjungwoo01 committed:
Add Checkstyle for code quality enforcement

This commit incorporates Checkstyle into the project to ensure adherence to coding standards and improve code quality. The integration of Checkstyle provides a systematic way to check the code for style deviations and potential errors. The following modifications have been made:

- Added `checkstyle.gradle` configuration file to define Checkstyle rules and setup.
- Updated `build.gradle` to include the Checkstyle plugin and its configuration, linking to the Checkstyle rules.
- Created `config/checkstyle/checkstyle.xml` to specify the coding standards and rules the project should follow.
- Adjusted existing code to comply with the newly integrated Checkstyle rules, ensuring the project passes all Checkstyle checks.

With Checkstyle integration, the project takes a significant step towards maintaining high code quality, consistency, and readability across all contributions.

hjungwoo01 committed:
Merge branch-A-CheckStyle to master

hjungwoo01 committed:
Add JavaFX and GUI Enhancements

This commit introduces JavaFX into the project, laying the foundation for a Graphical User Interface (GUI) and enhancing user interaction. The addition of JavaFX and GUI improvements aims to provide a more interactive and user-friendly experience. The following modifications have been implemented:

- Added JavaFX dependencies in the build.gradle file to support GUI development and integration.
- Created MainWindow.fxml and DialogBox.fxml to define the GUI layout, utilizing FXML for a more structured and manageable GUI design.
- Implemented MainWindow.java and DialogBox.java as controllers for the FXML files, facilitating dynamic interaction within the GUI.
- Modified Duke.java to initialize and launch the JavaFX application, integrating the backend logic with the frontend GUI.
- Updated the Ui class to support GUI operations, transitioning from console-based interaction to GUI-based feedback and user prompts.
- Added custom styling and layout adjustments to enhance the visual appeal and usability of the GUI.

By integrating JavaFX and focusing on GUI enhancements, this commit significantly improves the application's interactivity and accessibility, offering a modern and engaging user experience.

hjungwoo01 committed:
Add JavaFX and GUI Enhancements

This commit introduces JavaFX into the project, laying the foundation for a Graphical User Interface (GUI) and enhancing user interaction. The addition of JavaFX and GUI improvements aims to provide a more interactive and user-friendly experience. The following modifications have been implemented:

- Added JavaFX dependencies in the build.gradle file to support GUI development and integration.
- Created MainWindow.fxml and DialogBox.fxml to define the GUI layout, utilizing FXML for a more structured and manageable GUI design.
- Implemented MainWindow.java and DialogBox.java as controllers for the FXML files, facilitating dynamic interaction within the GUI.
- Modified Duke.java to initialize and launch the JavaFX application, integrating the backend logic with the frontend GUI.
- Updated the Ui class to support GUI operations, transitioning from console-based interaction to GUI-based feedback and user prompts.
- Added custom styling and layout adjustments to enhance the visual appeal and usability of the GUI.

By integrating JavaFX and focusing on GUI enhancements, this commit significantly improves the application's interactivity and accessibility, offering a modern and engaging user experience.

hjungwoo01 committed:
Enforce Checkstyle Coding Standards

This commit enforces Checkstyle coding standards on existing classes that did not previously comply with them. By doing so, we ensure that the entire codebase adheres to consistent coding conventions and improves code quality. The following changes have been made:
- Refactored and modified code where necessary to conform to Checkstyle standards.
- Verified that all affected classes now comply with Checkstyle coding standards.

Enforcing Checkstyle standards on existing code enhances code quality and consistency, making the project more maintainable and easier to collaborate on.

SherwynNg commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,97 @@
+            throw new DukeException("The description of a " + taskType + " cannot be empty.");
+        }
+
+        String description = commandParts[1].trim();
```
Do you think it will be better to create a method in Parser.java?

SherwynNg commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,97 @@
+            break;
+        case "event":
+            String[] eventParts = description.split(" /from ", 2);
+            if (eventParts.length < 2 || eventParts[1].trim().isEmpty()) {
+                throw new DukeException("The event time details or event times are missing.");
+            }
```
should these be extracted out to be a method on their own? To check for valid inputs

SherwynNg commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,23 @@
+import duke.task.Storage;
+import duke.task.TaskList;
+
+public class ExitCommand extends Command {
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+        return ui.showGoodbyeMessage();
+    }
+
+    @Override
+    public boolean isExit() {
+        return true;
+    }
+}
```
I like how you split the different commands into classes for better OOP. Good job!

SherwynNg commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,73 @@
+        if (filteredTasks.isEmpty()) {
+            return "No tasks found with the keyword: " + keyword;
+        } else {
+            String matchedTasks = IntStream.range(0, filteredTasks.size())
+                    .mapToObj(index -> (index + 1) + ". " + filteredTasks.get(index).toString())
+                    .collect(Collectors.joining("\n"));
```
I like how readable your code is, observing java coding standards.

SherwynNg reviewed (commented):
Everything looks good to me 👍 . 
Overall, your code is very readable because of the good OOP practices and coding standards.

yongkotaro commented on src/main/java/duke/task/Storage.java:
```diff
@@ -0,0 +1,147 @@
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the storage file and returns them as a TaskList.
+     *
+     * @return The TaskList containing tasks loaded from the storage file.
+     * @throws IOException If there is an issue reading the file.
+     */
```
I like that you have implemented JavaDocs for every method.

yongkotaro commented on src/main/java/duke/task/DateTimeUtil.java:
```diff
@@ -0,0 +1,51 @@
+
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.DukeException;
```
I appreciate the consistent ordering of import statements.

yongkotaro commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,73 @@
+     * @param fullCommand The full command string entered by the user.
+     */
+    public FindCommand(String fullCommand) {
+        String[] commandParts = fullCommand.split(" ", 2);
+        this.keyword = commandParts.length > 1 ? commandParts[1].trim() : "";
```
Maybe could use Parser to parse the command.

yongkotaro reviewed (commented):
Overall, the code followed the coding standard well and was easy to understand.

hjungwoo01 committed:
Integrate Assertions for Enhanced Code Reliability

This commit integrates Java assertions into various classes within the project to document important assumptions and enhance code reliability. By incorporating assertions, we assert essential preconditions, postconditions, and invariants that should hold at various points in the code. This practice helps in identifying potential bugs and ensuring that the code behaves as expected. The following modifications have been made:

Added assertions to Storage.java to validate file paths, ensure task list integrity, and verify file existence during read and write operations.
Implemented assertions in Parser.java to confirm the validity of user input commands and the non-nullity of command objects created from parsing.
Integrated assertions into AddCommand.java to check for the presence and correctness of task addition commands, along with the non-nullity of critical objects involved in executing these commands.
Incorporating assertions across these critical components improves the robustness of the application by catching and preventing erroneous states early in the development and testing phases. This enhancement makes the codebase more maintainable and aids developers in understanding and verifying the code's intended behavior.

hjungwoo01 committed:
Merge pull request #1 from hjungwoo01/branch-A-Assertions

Integrate Assertions for Enhanced Code Reliability

hjungwoo01 committed:
Improve Code Quality with Refactoring and Exception Handling Enhancements

This commit focuses on improving the overall code quality within the project through significant refactoring efforts and enhanced exception handling mechanisms. By refining the structure and clarity of our code, as well as more effectively managing exceptions, we aim to make our application more robust, readable, and maintainable. The following key changes have been implemented:

- Refactored the AddCommand.java class to simplify command execution logic, making it easier to understand and maintain. This includes breaking down complex methods into smaller, more manageable pieces and improving method naming for clarity.
- Enhanced exception handling in DateTimeUtil.java by adding meaningful logging within catch blocks. This approach aids in debugging and provides clearer insights into parsing errors without overwhelming the end user with technical details.
- Introduced clearer variable naming and simplified conditional statements across various classes, including Storage.java and Parser.java, to improve code readability and reduce cognitive load for future developers working on the project.

These improvements not only enhance the immediate readability and reliability of the code but also lay a foundation for more efficient future development. By addressing these aspects, we ensure that the application is easier to debug, extend, and maintain, thereby supporting a more agile and robust development lifecycle.

hjungwoo01 committed:
Merge pull request #2 from hjungwoo01/branch-A-CodeQuality

Improve Code Quality with Refactoring and Exception Handling Enhancements

hjungwoo01 committed:
Add Java Streams for Improved Efficiency

This commit focuses on enhancing the efficiency and readability of the codebase by leveraging Java streams in the FindCommand.java class. By incorporating streams, we streamline the process of searching for tasks based on a keyword within their descriptions. This approach not only improves the performance of the search operation but also makes the code more concise and expressive.

The use of streams allows for a more declarative and functional style of programming, enabling us to filter and manipulate collections of tasks with greater clarity and brevity. By adopting this approach, we aim to enhance the overall maintainability and comprehensibility of the codebase, facilitating easier understanding and future modifications.

These enhancements contribute to a more efficient and elegant solution for finding tasks within the Duke application, ultimately improving the user experience and developer productivity.

hjungwoo01 committed:
Merge pull request #3 from hjungwoo01/branch-A-Streams

Add Java Streams for Improved Efficiency

hjungwoo01 committed:
Add Chronological Sorting of Deadlines

This commit introduces an enhancement to the task listing functionality within the TaskList.java class of the Duke application. By implementing a method to sort deadlines chronologically, we significantly improve the user's ability to view upcoming tasks in an organized manner. This sorting ensures that deadlines are presented in the order of their occurrence, facilitating better planning and task management.

The addition of chronological sorting leverages a custom comparator within a sorting algorithm, focusing specifically on Deadline tasks. This method respects the original order of non-deadline tasks while bringing all deadline tasks to the forefront, sorted by their due dates. Such an approach not only enhances the usability of the listing feature but also maintains the integrity and structure of the task list.

Adopting this sorting mechanism contributes to a more intuitive and effective user experience. Users can now easily identify and prioritize tasks based on their deadlines, leading to improved task management and productivity. This update reflects our commitment to providing functional enhancements that address user needs while maintaining the simplicity and efficiency of the application's core functionalities.

Through this improvement, we aim to enrich the Duke application's feature set, making it more aligned with practical task management requirements and user expectations.

hjungwoo01 committed:
Merge pull request #4 from hjungwoo01/branch-C-Sort

Add Chronological Sorting of Deadlines

hjungwoo01 committed:
Improve JUnit Testing for Parser and AddCommand Classes

This commit  improves the JUnit testing coverage for the `Parser` and `AddCommand` classes within the Duke application. By extending the test cases to cover a wider array of input scenarios, including edge cases and potential error conditions, we have bolstered the robustness and reliability of the application's command parsing and task addition functionalities.

For the `Parser` class, additional tests have been implemented to rigorously evaluate the handling of unrecognized commands to ensure that the parser behaves as expected across all supported command types. This comprehensive testing strategy is designed to catch and prevent potential bugs that could arise from unexpected user inputs, enhancing the application's resilience against input-related errors.

Similarly, the `AddCommand` class now benefits from an expanded suite of tests that scrutinize the addition of todo, deadline, and event tasks under various conditions. This includes testing for tasks with extreme edge cases, such as very long descriptions or dates, and ensures proper exception handling for invalid task types or formats. By covering these additional scenarios, we ensure that task addition is handled gracefully and accurately, further improving the user experience.

These enhancements to the JUnit tests not only increase our confidence in the correctness of the application's core functionalities but also facilitate future development by providing a more robust framework for detecting and preventing regressions. This update underscores our commitment to quality and reliability, ensuring that the Duke application continues to meet and exceed user expectations for efficient and error-free task management.

Through meticulous testing and continuous improvement, we aim to maintain the Duke application's reputation for excellence, making it a dependable tool for users seeking an intuitive and effective task management solution.

hjungwoo01 committed:
Refactor all References from Duke to Nicky

This commit implements a comprehensive refactoring across the codebase, replacing all instances of "Duke" with "Nicky" to align with the project's renaming. This ensures consistency and coherence throughout the application, reflecting the new identity of the software as "Nicky".

hjungwoo01 committed:
Update README.md to Reflect Application Branding as Nicky

In this commit, the README.md file has been thoroughly updated to be compatible with the rebranding of the application to "Nicky". All references to the old name "Duke" have been meticulously replaced with "Nicky" to maintain uniformity across the documentation. This change solidifies the new branding within the user-facing components of the project and eliminates any potential confusion by ensuring that the documentation accurately represents the application's current state. The README now correctly guides users on setting up and interacting with Nicky, providing clarity and enhancing the overall user experience.

hjungwoo01 committed:
Adjust build.gradle for Nicky Compatibility

This commit updates the build.gradle file to align with the application's rebranding to "Nicky". All necessary dependencies, group names, and artifact ids previously associated with "Duke" have been changed to correspond with "Nicky". These modifications ensure that the build configurations and paths are in harmony with the application's new identity, facilitating a seamless build process and further embedding the Nicky brand into the project infrastructure.

hjungwoo01 committed:
Add JUnit Tests for Enhanced Command Functionality Verification

This commit introduces comprehensive JUnit tests for the `FindCommand` and `ListCommand` classes within the Nicky application, designed to rigorously verify the functionality of searching for tasks with specific keywords and listing all tasks in the application under various scenarios. The inclusion of these tests aims to ensure that the functionalities for searching and listing tasks operate correctly across different conditions, including the ability to accurately filter tasks by keywords and to provide users with an accurate and informative view of their current tasks list.

The primary purpose behind adding these tests is to maintain high code quality and reliability within the Nicky application. By thoroughly testing these key features, the project seeks to prevent regressions, enhance the application's stability, and ensure a seamless user experience for managing tasks. Additionally, these tests serve a critical role in documenting the expected behavior of the `FindCommand` and `ListCommand`, thereby facilitating easier understanding, maintenance, and future enhancements of the codebase.

For the `FindCommand`, the tests validate the command's capability to filter tasks based on the presence of a keyword in their descriptions, effectively handling scenarios where tasks either match or do not match the search criteria, including robust error handling for cases where no keyword is provided. Similarly, for the `ListCommand`, the tests confirm the command's functionality in listing tasks both when the task list is empty and when it contains multiple tasks, ensuring users receive accurate feedback on their task list status.

The implementation of these JUnit tests marks a significant step towards improving the Nicky application's code quality, underlining the project's commitment to delivering a dependable and user-friendly task management tool. These tests not only aid in early detection of potential issues during the development cycle but also act as a detailed documentation of the commands' intended behaviors, contributing to a more stable, maintainable, and reliable application.

hjungwoo01 committed:
Merge pull request #5 from hjungwoo01/branch-A-MoreTesting

Add JUnit Tests for Enhanced Command Functionality Verification

hjungwoo01 committed:
Enhance GUI Responsiveness for Dynamic Window Resizing

This commit significantly enhances the graphical user interface (GUI) of the Nicky application to support dynamic resizing of windows, ensuring that all visual components, including dialog boxes and their content, adapt fluidly to changes in the application window size. The focus of these improvements is to provide a more responsive and user-friendly interface that adjusts to varying display sizes and user preferences, thereby improving the overall usability and aesthetic appeal of the application.

The primary motivation behind these enhancements is to elevate the user experience by offering a GUI that is both flexible and adaptive. Recognizing that users may operate the application across devices with different screen sizes or simply prefer different window sizes, it became essential to implement a GUI layout that responds to these variations seamlessly. By making the dialog boxes and other UI elements dynamically resize with the window, the application now better accommodates user needs, ensuring that the display of messages, inputs, and other information remains optimal under various conditions.

Key aspects of the GUI, such as the dialog containers and scroll panes, were meticulously adjusted to ensure that they not only resize horizontally but also maintain their functional integrity, such as proper scrolling behavior and content visibility. This ensures that users can interact with the application comfortably, without experiencing layout issues or content accessibility problems, regardless of the window size.

The enhancements to the GUI's responsiveness underscore the project's dedication to providing a high-quality user experience, demonstrating a commitment to user-centric design principles. These improvements make the Nicky application more accessible and enjoyable to use, reflecting an ongoing effort to adapt to user feedback and technological advancements.

By implementing these dynamic resizing features, the Nicky application takes a significant step forward in usability and design sophistication, reinforcing its position as a reliable and user-friendly task management tool. This initiative not only addresses current user needs but also lays a foundation for future enhancements, ensuring that the application continues to meet and exceed user expectations in an ever-evolving digital landscape.

hjungwoo01 committed:
Merge pull request #6 from hjungwoo01/branch-A-BetterGui

Enhance GUI Responsiveness for Dynamic Window Resizing

hjungwoo01 committed:
Develop Comprehensive User Guide for Enhanced User Onboarding

This commit introduces a detailed User Guide for the Nicky application, aimed at significantly improving the user onboarding experience. The guide covers key functionalities such as listing tasks, adding various types of tasks (todos, deadlines, events), finding tasks, and marking tasks as done or undone. The inclusion of clear instructions, action descriptions, usage examples, and expected outcomes for each feature ensures that users can easily understand and utilize the application to its full potential right from the start.

The primary motivation behind creating this User Guide is to lower the entry barrier for new users and enhance the overall usability of the application. Recognizing that first-time users may not be familiar with the command-line interface and the specific syntax required for different operations, it became essential to provide a comprehensive resource that users can refer to when interacting with Nicky. By detailing each feature with examples and expected outcomes, the guide aims to empower users to efficiently manage their tasks without needing to experiment with commands or syntax.

Moreover, the User Guide serves as a quick reference that can help users refresh their memory on specific commands and functionalities, thereby enhancing long-term user engagement and satisfaction. It also reflects the project's commitment to user-centric design and support, highlighting the importance placed on clear communication and ease of use.

The creation of this User Guide marks a pivotal step towards making the Nicky application more accessible, user-friendly, and intuitive. This effort not only aids in user retention by providing immediate assistance and guidance but also sets a foundation for future updates and features, ensuring that the application continues to evolve in ways that meet and exceed user expectations in the dynamic landscape of task management tools.

hjungwoo01 committed:
Rename DaDuke Image to DaNicky

Renamed the application image from 'DaDuke.png' to 'DaNicky.png' to reflect the rebranding of the application and ensure consistency with the new project identity.

hjungwoo01 committed:
Merge pull request #7 from hjungwoo01/branch-A-UserGuide

Develop Comprehensive User Guide for Enhanced User Onboarding

hjungwoo01 committed:
Add UI Image to README and Update Nicky to Version v0.2

Included a UI image in the README.md file to provide a visual overview of the Nicky application interface. This enhancement aims to improve documentation and assist users in understanding the application's layout and functionalities at a glance. Additionally, updated the Nicky application to version v0.2, incorporating new features, bug fixes, and performance improvements to enhance the overall user experience and application stability.

hjungwoo01 committed:
Merge pull request #8 from hjungwoo01/branch-A-UserGuide

Add UI Image to README and Update Nicky to Version v0.2

hjungwoo01 committed:
Update Javadoc to adhere to coding standard

Updated Javadoc comments to adhere to coding standard format, ensuring
the phrasing of the overview statement conforms appropriately.

Adhering to the coding standard ensures consistency and readability
across the codebase, making maintenance and collaboration more
efficient.

hjungwoo01 committed:
Merge branch 'master' of https://github.com/hjungwoo01/ip
